runtime/compiler/optimizer/InlinerTempForJ9.cpp

```diff
@@ -4780,16 +4780,23 @@ TR_J9InlinerPolicy::supressInliningRecognizedInitialCallee(TR_CallSite* callsite
          return true;
       case TR::java_lang_Class_cast:
          return true; // Call will be transformed into checkcast
+      case TR::java_lang_Class_isHidden:
+      case TR::java_lang_Class_isAnonymousClass:
+      case TR::java_lang_Class_isEnum:
+      case TR::java_lang_Class_isSynthetic:
+      case TR::java_lang_Class_isArray:
+      case TR::java_lang_Class_isPrimitive:
+      case TR::java_lang_Class_isAnnotation:
       case TR::java_lang_Object_hashCode:
-         if (callNode &&
-             callNode->getFirstChild() &&
-             callNode->getFirstChild()->getOpCode().hasSymbolReference())
+         if (callNode
+             && callNode->getNumChildren() > 0)
```

Comment:
Maybe more has been deleted here than was intended ? I think it is okay to check that the number of children is > 0 but did you also intend to avoid checking if the first (or maybe it is last since that was changed in the last commit I think ?) child has a symbol reference before accessing it ?

Reply:
I re-added checking if the last child has symref before accessing it, now that I realize that `getSymbolReference` would fail an assertion instead of just returning NULL.

Reply:
I have also removed `java/lang/Class.isInstance` from here since the decision to inline that is done during ILGen (through native call inlining). I will contribute that check to skip inlining that as well as the VP transformation in a separate PR.